marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowImpl.java

```diff
@@ -97,18 +97,18 @@ public String getName() {
         return this.name;
     }
 
-    public void setName(String flowName) {
-        this.name = flowName;
-    }
-
     @Override
     public String getId() {
         return id;
     }
 
     @Override
     public void setId(String id) {
-        this.id = id;
+        this.id = id == null ? UUID.randomUUID().toString() : id;
```

Comment:
Don't need ID field as flowId == flowName
That is how we search for the Flow in the filesystem.